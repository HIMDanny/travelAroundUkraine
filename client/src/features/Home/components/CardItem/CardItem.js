import React, { memo } from 'react';
import PropTypes from 'prop-types';
import { Typography, Box, styled } from '@mui/material';
import ArrowForwardIosIcon from '@mui/icons-material/ArrowForwardIos';
import { useNavigate } from 'react-router-dom';

const CardItemWrapper = styled('div')({
  minWidth: '265px',
  maxWidth: '265px',
  height: '400px',
  backgroundSize: 'cover',
  color: 'white',
  position: 'relative',
  cursor: 'pointer',

  '&::after': {
    content: "''",
    position: 'absolute',
    width: '100%',
    height: '100%',
    background: 'rgba(0,0,0,0.5)',
    left: '0',
    top: '0',
    transition: 'opacity 0.5s',
    opacity: '1',
  },

  '&:hover::after': {
    opacity: '0',
    cursor: 'pointer',
  },
});

const CardItemPrice = styled(Typography)(({ theme }) => ({
  backgroundColor: theme.palette.primary.main,
  textAlign: 'center',
  width: '108px',
  fontSize: '18px',
  paddingTop: '19px',
  paddingBottom: '19px',
  fontWeight: 'bold',
  letterSpacing: '1px',
  marginTop: '0',
  position: 'absolute',
  zIndex: '1',
  color: theme.palette.common.white,
}));

const CardItemHeader = styled((props) => <Typography variant="h3" {...props} />)(({ theme }) => ({
  position: 'absolute',
  marginTop: '250px',
  paddingLeft: '25px',
  paddingRight: '25px',
  zIndex: '2',
  textTransform: 'uppercase',
  color: theme.palette.common.white,
}));

const CardItemLocation = styled('div')({
  height: '20px',
  display: 'flex',
  paddingLeft: '25px',
  paddingRight: '25px',
  zIndex: '2',
  position: 'absolute',
  width: '215px',
  alignItems: 'center',
  marginTop: '350px',
});

const CardItemText = styled(Typography)(({ theme }) => ({
  margin: '0',
  whiteSpace: 'nowrap',
  color: theme.palette.common.white,
}));

const CardItem = ({ name, region, currentPrice, imageUrls, itemNo }) => {
  const navigate = useNavigate();
  const url = imageUrls[0];
  //   console.log(url);

  return (
    <CardItemWrapper
      onClick={() => navigate(`/tour/${+itemNo}`)}
      sx={{
        height: '400px',
        backgroundImage: `url(${url})`,
      }}
    >
      <CardItemPrice>{currentPrice} $</CardItemPrice>
      <CardItemHeader>{name}</CardItemHeader>
      <CardItemLocation>
        <Box
          component="img"
          src="https://visitukraine.today/assets/img/fromukraine/maps-and-flags.png"
          alt="location"
          sx={{ paddingRight: '10px', maxWidth: 'min-content' }}
        />
        <CardItemText>{region} region</CardItemText>
        <ArrowForwardIosIcon sx={{ display: 'block', marginLeft: 'auto' }} fontSize="small" />
      </CardItemLocation>
    </CardItemWrapper>
  );
};

CardItem.propTypes = {
  name: PropTypes.string.isRequired,
  region: PropTypes.string.isRequired,
  currentPrice: PropTypes.number.isRequired,
  imageUrls: PropTypes.array.isRequired,
  itemNo: PropTypes.string.isRequired,
};

export default memo(CardItem);
