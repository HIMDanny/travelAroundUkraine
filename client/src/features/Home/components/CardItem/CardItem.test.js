import React from 'react';
import { render } from '@testing-library/react';
import CardItem from './CardItem';

jest.mock('@mui/material/Typography', () => ({ children }) => <h2>{children}</h2>);
jest.mock('@mui/material/Box', () => ({ children }) => <img alt="test img" />);

describe('Card Item Snapshot test', () => {
  test('should CardItem match snapshot', () => {
    const { asFragment } = render(<CardItem />);
    expect(asFragment()).toMatchSnapshot();
  });
});
