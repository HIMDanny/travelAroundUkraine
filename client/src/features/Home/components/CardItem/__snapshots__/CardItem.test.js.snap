// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Item Snapshot test should CardItem match snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-1iryrks"
  >
    <h2>
      $ 200.00
    </h2>
    <h2>
      Kyiv city walk
    </h2>
    <div
      class="css-h56svl"
    >
      <img
        alt="test img"
      />
      <h2>
        Kyiv region
      </h2>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-bwo5fi-MuiSvgIcon-root"
        data-testid="ArrowForwardIosIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M6.23 20.23 8 22l10-10L8 2 6.23 3.77 14.46 12z"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;
